Compilation fails when profiling disabled and CUDA enabled

Both master and develop branches fail to compile if `Kokkos_ENABLE_PROFILING` is disabled. This happens when compiling the _CUDA backend, using Clang compiler_. Error `DeviceTypeTraits` is not declared, I guess due to not including `Kokkos_Profiling_Interface.hpp` that declares it...

```
In file included from /home/josip/Code/kokkos/core/src/Cuda/Kokkos_Cuda_Locks.cpp:51:
/home/josip/Code/kokkos/core/src/Kokkos_Cuda.hpp:279:8: error: explicit specialization of undeclared template struct 'DeviceTypeTraits'
struct DeviceTypeTraits<Cuda> {
       ^               ~~~~~~
/home/josip/Code/kokkos/core/src/Kokkos_Cuda.hpp:281:20: error: unknown type name 'DeviceType'
  static constexpr DeviceType id = DeviceType::Cuda;
```

--- a/.jenkins
+++ b/.jenkins
@@ -95,6 +95,7 @@ pipeline {
                                 -DKokkos_ENABLE_TESTS=ON \
                                 -DKokkos_ENABLE_CUDA=ON \
                                 -DKokkos_ENABLE_CUDA_LAMBDA=ON \
+                                -DKokkos_ENABLE_PROFILING=OFF \
                               .. && \
                               make -j8 && ctest --output-on-failure'''
                     }

--- a/core/src/Kokkos_CudaSpace.hpp
+++ b/core/src/Kokkos_CudaSpace.hpp
@@ -56,6 +56,8 @@
 
 #include <Kokkos_HostSpace.hpp>
 
+#include <impl/Kokkos_Profiling_Interface.hpp>
+
 #include <Cuda/Kokkos_Cuda_abort.hpp>
 
 #ifdef KOKKOS_IMPL_DEBUG_CUDA_PIN_UVM_TO_HOST

